File: leetcode/golang/83.删除排序链表中的重复元素.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return head;
    }

    start := head;
    for start.Next != nil {
        if start.Val == start.Next.Val {
            start.Next = start.Next.Next;
        } else {
            start = start.Next;
        }
    }
    return head;
}

File: leetcode/python/83.删除排序链表中的重复元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if not head or not head.next:
            return head
        
        start = head
        while start.next:
            if start.val == start.next.val:
                start.next = start.next.next
            else:
                start = start.next
        
        return head